enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -837,8 +837,21 @@ func (s *ExecutionServer) dispatch(ctx context.Context, req *repb.ExecuteRequest
 		return nil, status.WrapError(err, "get executor pool info")
 	}
 	var hostnamePattern string
+	var routingConfig *scpb.RoutingConfig
 	if exp := s.env.GetExperimentFlagProvider(); exp != nil {
+		// TODO: delete this once we're only using routing_config.
 		hostnamePattern = exp.String(ctx, "remote_execution.executor_hostname_pattern", "")
+		const routingConfigExperimentName = "remote_execution.executor_routing_config"
+		routingConfigObj, details := exp.ObjectDetails(ctx, routingConfigExperimentName, nil)
+		if len(routingConfigObj) > 0 {
+			routingConfig = &scpb.RoutingConfig{}
+			if err := experiments.ObjectToProto(routingConfigObj, routingConfig); err != nil {
+				alert.CtxUnexpectedEvent(ctx, "Error converting routing config object %+#v to proto: %s", routingConfigObj, err)
```

Comment:
You shouldn't put the raw object into the event name as that gets fed as a label value. Move it to the message.